### PR TITLE
Making Abstract abstract, instead of introduction

### PR DESCRIPTION
**Pull request by TallTed**
pulled from #1554

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1560.html" title="Last updated on Jan 23, 2025, 6:35 PM UTC (d8e8d4e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1560/a18b601...d8e8d4e.html" title="Last updated on Jan 23, 2025, 6:35 PM UTC (d8e8d4e)">Diff</a>

### CODE REVIEW

**Review by decentralgabe — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -267,12 +267,20 @@
...
       <p>
-[=Credentials=] are integral to our daily lives; driver's licenses confirm
-our capability to operate motor vehicles, university degrees assert our level
-of education, and government-issued passports permit travel between countries.
-This specification provides a mechanism to express these sorts of
-[=credentials=] on the Web in cryptographically secure, privacy-respecting,
-and machine-verifiable way.
+This specification provides normative and non-normative guidance on
+implementing and managing [=verifiable credentials=] and associated
+cryptographic practices. It emphasizes the importance of understanding and
```
Comment (iherman) [inline index.html:272]:
```suggestion
This specification provides a standard and non-normative guidance on
implementing and managing [=verifiable credentials=] and associated
cryptographic practices. It emphasizes the importance of understanding and
```

I am not sure about the "guidance" term here. This is a standard, and the normative items should be emphasized as such.

Comment (TallTed) [reply index.html:272]: @iherman — Does this cover your concern?
```suggestion
This specification provides normative requirements and non-normative guidance
for implementing and managing [=verifiable credentials=] and associated
cryptographic practices. It emphasizes the importance of understanding and
```

Comment (iherman) [reply index.html:272]: Yes, thank you.

**Review by iherman — APPROVED** [review]
Approving after the merge of https://github.com/w3c/vc-data-model/pull/1560#discussion_r1757293580

**Review by msporny — CHANGES_REQUESTED** [review]
I don't find this change to be an improvement on the first paragraph that someone that doesn't know about the specification would find helpful for the following reasons:

* It doesn't explain what a verifiable credential is (it introduces the concept too early)
* It starts talking about cryptography without making it clear why it matters
* It injects jargon like "cryptoperiod" into the second sentence the reader reads
* It makes no mention of the data model or what we're trying to accomplish with the specification
* It skips over most of the middle of the spec and then starts identifying things we talk about in the security and privacy considerations.

I don't think we do the readers a favor in exposing them to this paragraph as the first thing they read about the specification. I know that I would get a bad impression of the specification if this were the first thing I read about it. 

I do think it would be useful to have this text in a "This is what this specification contains" with pointers to each section. That would go well in the introduction, so that's not to say that we shouldn't include this text, just not as the very first thing that someone reads.

I'll also note that the Abstract does need work; it's dated and could use a bit of "oomph" in the last sentence or two.

IOW, keep the first two sentences, I like them. The last sentence isn't great:

> These credentials provide benefits to us when used in the physical world, but their use on the Web continues to be elusive.

We need to modernize that statement, and we probably have space for one or two more high-level but impactful statements regarding the vision of the technology.

An abstract "lets readers get the gist or essence of the paper quickly, in order to decide whether to read the full paper" -- I don't think we should do that by summarizing the documents contents. We should achieve that by clearly articulating the vision of the technology... what does the world look like now, and how does this specification improve upon the state of the world.

We're trying to say: "This technology enables people to take their physical credentials into the online world but with better security and utility than traditional physical credentials." ... without diving down into the details. If people read just that first paragraph, they should get a good idea of what the technology is about.

**Comment by TallTed** [comment]
I do not think that the existing "Abstract" *is* an abstract, and I would rather see *no* titular "Abstract" than have it be persisted here as such, especially given that its text is also found _in toto_ as the first paragraph of the "Introduction".

I can see no justification for using the same paragraph both as the document's "Abstract" *and* as the first paragraph of the "Introduction". I have found no example of similar in any other W3 spec (and if such surfaces, I would strongly advise _that_ WG or its successor to change it).

An abstract ("[a short form of a speech, article, book, etc., giving only the most important facts or ideas](https://dictionary.cambridge.org/us/dictionary/english/abstract)") has a different purpose than an introduction ("[a short speech or piece of writing that comes before a longer speech or written text, usually giving basic information about what is to follow](https://dictionary.cambridge.org/us/dictionary/english/introduction)").

In my experience of Abstracts in other fields, and even in this field outside of W3C specifications, they generally assume that their readers are more than passingly familiar with the subject matter. Perhaps this is because they are largely found in academic papers that are destined for a conference of subject matter experts.

I will readily grant that my suggested abstraction is imperfect. I said when I first suggested it that I used ChatGPT to produce the start of it, though I did make some edits to what ChatGPT provided (which I regrettably did not preserve in its original form).

> We're trying to say: "This technology enables people to take their physical credentials into the online world but with better security and utility than traditional physical credentials."

I think it's overly simplified, but that quoted sentence is better than the existing "Abstract", and I would not object to using a slightly adjusted version of that quoted sentence instead of my big paragraph:

> The technology specified herein enables people to take analogs of their traditional physical credentials into the online world, while providing better security and utility in the new online credentials than may be found in the old.

**Comment by msporny** [comment]
@TallTed wrote:
> I can see no justification for using the same paragraph both as the document's "Abstract" and as the first paragraph of the "Introduction". 

I agree, so can we change the introduction to something else?

> they generally assume that their readers are more than passingly familiar with the subject matter. Perhaps this is because they are largely found in academic papers that are destined for a conference of subject matter experts.

Yes, and I find this approach in academic literature quite obtuse (and lazy) and is one of the reasons why many scientists struggle with conveying why their ideas are important to the world. This is a global standard, that's open to the public. The public, with a high-school level of education, should be able to read our Abstract and understand what the document is about. I'd go a bit further and argue that they should be able to read just about everything in the Introduction and get a good idea of what this ecosystem is about and how we achieve what we achieve (and we're increasingly failing to do this in the specification, IMHO).

> The technology specified herein enables people to take analogs of their traditional physical credentials into the online world, while providing better security and utility in the new online credentials than may be found in the old.

^ Yes, but that's what the current Abstract text is basically saying, isn't it... only, in a way that is better than the text above (which is an improvement, in some areas, and not in others -- it's too vague, IMHO).

I'd like to see more opinions on what we should try to do here. I think we need more than a sentence, but less than what's been proposed. Three to four sentences feels adequate (which is kinda-sorta what we have now), with some word smithing.

**Comment by wip-abramson** [comment]
I tend to agree with @msporny on this. 

I think the current abstract is okay, but there is room for some minor wordsmithing.

I agree with @TallTed that the abstract should not be repeated in the introduction. So we should remove that text from the introduction and propose some new text there that introduces the sections and content of the specification. I think @TallTed the paragraph you propose in this PR is in that direction.

File: index.html [hunk 2]
```diff
@@ -267,12 +267,20 @@
...
     <section id='abstract'>
       <p>
-[=Credentials=] are integral to our daily lives; driver's licenses confirm
-our capability to operate motor vehicles, university degrees assert our level
-of education, and government-issued passports permit travel between countries.
-This specification provides a mechanism to express these sorts of
-[=credentials=] on the Web in cryptographically secure, privacy-respecting,
-and machine-verifiable way.
+This specification provides normative requirements and non-normative guidance
+for implementing and managing [=verifiable credentials=] and associated
```
Comment (wip-abramson) [inline index.html:271]: I prefer `expressing` over `implementing and managing`. I think it aligns more with the fact that this spec is a for a data model.

Comment (TallTed) [reply index.html:271]: @wip-abramson — Maybe this?
```suggestion
for expressing and consuming [=verifiable credentials=] and associated
```

File: index.html [hunk 3]
```diff
@@ -267,12 +267,20 @@
...
+cryptographic practices. It emphasizes the importance of understanding and
+updating cryptographic systems and managing private signing keys with limited
+cryptoperiods. It discusses mechanisms for ensuring content integrity of
```
Comment (brentzundel) [inline index.html:274]:
```suggestion
updating cryptographic systems and managing private signing keys with limited
validity periods. It discusses mechanisms for ensuring content integrity of
```

**Review by brentzundel — CHANGES_REQUESTED** [review]
minor nit, otherwise LGTM

**Review by selfissued — APPROVED** [review]

**Comment by msporny** [comment]
No forward movement on this PR in many weeks. Marking it as pending close.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2025-01-15-vcwg#section1-5) on 2025-01-15

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.5. Making Abstract abstract, instead of introduction (pr vc-data-model#1560)

_See github pull request [vc-data-model#1560](https://github.com/w3c/vc-data-model/pull/1560)._




**Brent Zundel:** this PR is about making the abstract abstract.  

**Manu Sporny:** TallTed I think you objected because the content matched the introduction.  
… I think I objected to your proposed text because it was too wordy. It would be good to make an attempt at it.  
… I think the current PR is not going to go it. I marked it pending close.  

**Ted Thibodeau Jr.:** As I put on the PR. The abstract that chatgpt helped me make is an abstract of the entire document.  
… The current abstract is introductory. And is not an abstract.  
… I don't see value having the current paragraph as an abstract.  
… I suggest dropping it an not having an abstract.  

**Brent Zundel:** One path would be to modify it to remove the paragraph. The other path would be to make it more concise.  

**Manu Sporny:** I dont think it improves the document in its current shape.  
… I dont think we can delete the abstract, because respec complains.  
… I would be fine with keeping the abstract and reworking the introduction spec.  
… Typically the W3C abstracts dont follow the academic abstract pattern.  
… They aim to give folks a paragraph concise understanding of the spec.  
… Suggest we try to keep reworking it until it is acceptable.  
… I will make another attempt.  

**Ivan Herman:** dropping the abstract is not an option due to the publishing rules.  

**Brent Zundel:** Those are the PRs in the VCDM.  
… manu do you have the set of at risk sections.  

**Manu Sporny:** bigbluehat might have those, I dont think we have any more.  

**Benjamin Young:** Yep, not in the VCDM.  
… bitstring has a few things.  

</details>

**Review by iherman — COMMENTED** [review]
I do not want to get into the "philosophical" discussion on what the abstract is. The new text's goals, size, etc, would fulfill what I expect from an abstract.

Except that... the current text is terribly unbalanced. Apart from the introduction sentence, it almost exclusively refers to issues, texts, etc., from the privacy and security sections. Those are of course important, but there is no mention of the fact that the specification defines a data model, that this model is the basis for a number of other recommendations that, together, make up verifiable credentials, that it defines a number of common terms to express credentials in an application independent way, etc., etc. All in all, it does not really give a coverage of the specification as a whole.

I am not a big fan of the original text but, I believe, it is still better than this text.

My preference is to close this PR without merging.

**Comment by David-Chadwick** [comment]
@iherman Whilst I agree with your sentiments, rather I would propose fixing the text rather than deleting it

**Review by David-Chadwick — CHANGES_REQUESTED** [review]

File: index.html [hunk 4]
```diff
@@ -267,12 +267,20 @@
   <body>
     <section id='abstract'>
       <p>
-[=Credentials=] are integral to our daily lives; driver's licenses confirm
-our capability to operate motor vehicles, university degrees assert our level
-of education, and government-issued passports permit travel between countries.
-This specification provides a mechanism to express these sorts of
-[=credentials=] on the Web in cryptographically secure, privacy-respecting,
-and machine-verifiable way.
+This specification provides normative requirements and non-normative guidance
```
[the remark this answers is not in the document]
Comment (iherman) [reply index.html:270]: I like that, thank you.

I would also reduce a bit the subsequent text, focussed on detailed security issues, but I lack the knowledge to know the priorities there.

[the remark this answers is not in the document]
Comment (msporny) [reply index.html:270]: I'd be fine with @David-Chadwick's suggestions, with the following changes applied:

* Give an example of what a verifiable credential is to the reader.
* Remove the "must contain and may contain" language and replace it with "can contain".
* Specify what the three parties in the model are: issuer, holder, verifier.
* Clean up the grammatical flow between the "It specifies/It is/It also" repetition that start each sentence.

Delete or summarize the rest as it detracts from the main takeaway.

File: index.html [hunk 5]
```diff
@@ -267,12 +267,20 @@
...
+validity periods. It discusses mechanisms for ensuring content integrity of
+linked external resources and highlights the risks of unsigned claims.
+Strategies are provided to mitigate man-in-the-middle (MITM), replay, and
+spoofing attacks, and to address issues related to credential atomization,
+validity periods, and device security. This specification also covers
+acceptable use of credentials, warns against code injection risks, and
+underscores the need for accessibility and internationalization
+considerations, advocating for a data-first approach and adherence to
+internationalization standards to ensure correct rendering of
+multilingual text.
```
Comment (msporny) [inline index.html:283]:
```suggestion
A [=verifiable credential=] is a digital representation of a set of [=claims=] made 
by an [=issuer=], such as a driver's license or a education certificate. This 
specification describes an extensible data model for [=verifiable credentials=], 
how they can be secured from tampering, and a three-party ecosystem that 
exchanges these credentials that is composed of [=issuers=], [=holders=], 
and [=verifiers=]. This document also covers a variety of security, privacy,
internationalization and accessibility considerations for ecosystems that 
use the technologies described in this specification.
```

Comment (msporny) [reply index.html:283]: Here is a proposal for an abstract rewrite that I would be a +1 to merge (with @TallTed's inevitable change suggestions that improve the text).

**Review by msporny — CHANGES_REQUESTED** [review]

File: index.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (msporny) [reply index.html:270]: Concrete change suggestion here: https://github.com/w3c/vc-data-model/pull/1560#discussion_r1921127205

File: index.html (hunk 5 above)
Comment (dlongley) [reply index.html:283]: I believe the above suggestion should say "A [=verifiable credential=] is a specific way to express a set of [=claims=]". VCs can be printed / embedded physically and even if the information itself remains digital, this might confuse someone in an abstract.

Comment (msporny) [reply index.html:283]: @dlongley, applied in the change suggestion above.

**Comment by iherman** [comment]
I seem to be lost in the different changes. The text in https://github.com/w3c/vc-data-model/pull/1560#pullrequestreview-2560505494 is pretty different from https://github.com/w3c/vc-data-model/pull/1560#discussion_r1921127205 and its related comments. 

Specifically, the following text has no equivalent:

> This specification describes the extensible data model for [=verifiable credentials=], 
how they can be secured from tampering, and a three-party ecosystem that 
exchanges these credentials that is composed of [=issuers=], [=holders=], 
and [=verifiers=]. 

Not to even mention the term "data model" in the abstract of a specification whose title includes the term "Data Model" is kind of strange to me. Has there been an omission somewhere?

**Comment by msporny** [comment]
> I seem to be lost in the different changes. The text in [#1560 (review)](https://github.com/w3c/vc-data-model/pull/1560#pullrequestreview-2560505494) is pretty different from [#1560 (comment)](https://github.com/w3c/vc-data-model/pull/1560#discussion_r1921127205) and its related comments.

Yes, I was asked to provide a proposal on something that I could +1. I read David's changes as well and tried to incorporate them. After multiple re-writes, the text above is what I landed on. It incorporates the portions of the original text, and David's text, that it feels like we should keep. 

> Not to even mention the term "data model" in the abstract of a specification whose title includes the term "Data Model" is kind of strange to me. 

Why is it strange to use the term "data model" when describing a specification about a data model?

> Has there been an omission somewhere?

I don't think so? I think you might be thinking that the text I'm proposing is a minor re-write of the proposed text in the PR. It's not, it's a significant re-write, but one that tries to include the concepts worth keeping in the original text and removes the details that are unnecessary in the final text.

Does that help?

**Comment by dlongley** [comment]
> I seem to be lost in the different changes. The text in [#1560 (review)](https://github.com/w3c/vc-data-model/pull/1560#pullrequestreview-2560505494) is pretty different from [#1560 (comment)](https://github.com/w3c/vc-data-model/pull/1560#discussion_r1921127205) and its related comments.

Yeah, I think the way github is rendering the page is leading to confusion. The first link to the review by @msporny does not include his suggestion, which is actually further up the page. All it does is quote what's currently in the PR -- which is the text @msporny's suggestion (again, only seen above) changes.

> Specifically, the following text has no equivalent:
> 
> > This specification describes the extensible data model for [=verifiable credentials=],
> > how they can be secured from tampering, and a three-party ecosystem that
> > exchanges these credentials that is composed of [=issuers=], [=holders=],
> > and [=verifiers=].
> 
> Not to even mention the term "data model" in the abstract of a specification whose title includes the term "Data Model" is kind of strange to me. Has there been an omission somewhere?

@msporny's suggestion above is what would include the text above that talks about "the extensible data model". My read of your comment here is different from @msporny's ... I think you're saying that the abstract SHOULD mention "data model" (as @msporny's suggestion does and you quote from it here) and I think he read this as you think it SHOULD NOT. Can you clarify your meaning?

To be extra clear, @msporny's suggestion for the abstract is here: https://github.com/w3c/vc-data-model/pull/1560#discussion_r1921127205

And had this text at the time of my writing this comment:

```
A [=verifiable credential=] is a specific way to express a set of [=claims=] 
made by an [=issuer=], such as a driver's license or an education certificate. This 
specification describes the extensible data model for [=verifiable credentials=], 
how they can be secured from tampering, and a three-party ecosystem that 
exchanges these credentials that is composed of [=issuers=], [=holders=], 
and [=verifiers=]. This document also covers a variety of security, privacy,
internationalization, and accessibility considerations for ecosystems that 
use the technologies described in this specification.
```

I'm supportive of this text and it addresses my comment about VCs being potentially expressed on physical documents, etc.

**Comment by msporny** [comment]
> Yeah, I think the way github is rendering the page is leading to confusion. 

Oh, I didn't see that weird rendering until now... I had to reload my page... what is going on, Github!?

**Comment by iherman** [comment]
> > Yeah, I think the way github is rendering the page is leading to confusion.
> 
> Oh, I didn't see that weird rendering until now... I had to reload my page... what is going on, Github!? Looks like a Github bug?

Yeah, that was a bit (a lot!) confusing. 

To make things clear from my point of view, the text of @dlongley in his latest comment which, to be sure, I copy here verbatim:

> A [=verifiable credential=] is a specific way to express a set of [=claims=] 
made by an [=issuer=], such as a driver's license or an education certificate. This 
specification describes the extensible data model for [=verifiable credentials=], 
how they can be secured from tampering, and a three-party ecosystem that 
exchanges these credentials that is composed of [=issuers=], [=holders=], 
and [=verifiers=]. This document also covers a variety of security, privacy,
internationalization, and accessibility considerations for ecosystems that 
use the technologies described in this specification.

works for me. And yes, @dlongley, my intention was that the data model SHOULD be mentioned. Which is in the text above.

File: index.html (hunk 5 above)
Comment (David-Chadwick) [reply index.html:283]: Please reword the following
"and a three-party ecosystem that  exchanges these credentials that is composed of [=issuers=], [=holders=], 
and [=verifiers=]. "
to
"and a three-party ecosystem for the exchange of these credentials that is composed of [=issuers=], [=holders=], 
and [=verifiers=]. "

**Review by TallTed — COMMENTED** [review]
I had these two suggestions pending against my own PR.

It appears there are new suggested abstracts from both @David-Chadwick and @msporny, each with some changes requested against them.

I would also have some tweaks for both, before I would merge them into this PR ... but it would be better to focus on one, I think.

**Comment by David-Chadwick** [comment]
Can I suggest that @msporny and myself privately agree on a revised text and then post it to this PR for everyone else to comment on

**Comment by dlongley** [comment]
Trying to help out...

If you guys (@msporny and @David-Chadwick) both agree on this text (which includes edits from David's comment [here](https://github.com/w3c/vc-data-model/pull/1560#discussion_r1923432170)), let's use this:

> A [=verifiable credential=] is a specific way to express a set of [=claims=] 
made by an [=issuer=], such as a driver's license or an education certificate. This 
specification describes the extensible data model for [=verifiable credentials=], 
how they can be secured from tampering, and a three-party ecosystem for 
the exchange of these credentials that is composed of [=issuers=], [=holders=], 
and [=verifiers=]. This document also covers a variety of security, privacy,
internationalization, and accessibility considerations for ecosystems that 
use the technologies described in this specification.

I believe this is the text that was agreed to by participants on the call (which included Manu), provided that it included David's change, which the above text does. So if @msporny and @David-Chadwick both agree to this text, let's use this.

**Comment by David-Chadwick** [comment]
@dlongley Yes lets go with this

**Comment by iherman** [comment]
Let us use this and call it a day :-)
